packages/amplify_datastore_plugin_interface/lib/src/types/temporal/datetime_parse.dart

```diff
@@ -1,22 +1,27 @@
-/*
- * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License").
- * You may not use this file except in compliance with the License.
- * A copy of the License is located at
- *
- *  http://aws.amazon.com/apache2.0
- *
- * or in the "license" file accompanying this file. This file is distributed
- * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
- * express or implied. See the License for the specific language governing
- * permissions and limitations under the License.
- */
-
-import 'package:date_time_format/date_time_format.dart';
-
-String formatDateToISO8601(DateTime dateTime) {
-  return dateTime == null
-      ? null
-      : DateTimeFormat.format(dateTime, format: DateTimeFormats.iso8601);
-}
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:date_time_format/date_time_format.dart';
+
+// ignore: public_member_api_docs
+extension DateTimeParse on DateTime {
+  String toDateTimeIso8601String() {
+    return DateTimeFormat.format(this, format: DateTimeFormats.iso8601);
+  }
+
+  static DateTime fromString(String input) {
+    return input == null ? null : DateTime.parse(input);
+  }
```

Comment:
Unit tests for these. These have been changed enough times that I don't feel comfortable shipping without tests to nail down the input and output contract and expectations.